patches/uboot/0002-arm-rpi-copy-rpi-fw-modified-fdt-settings-to-u-boot-.patch

```diff
@@ -0,0 +1,331 @@
+From 1f0277187d788f6def4a5e12340efdcddbcc6fc2 Mon Sep 17 00:00:00 2001
+From: Leonardo Amorim <lam@vizrt.com>
+Date: Thu, 7 Apr 2022 17:01:17 -0300
+Subject: [PATCH] arm: rpi: copy rpi fw modified fdt settings to u-boot loaded
+ fdt
+
+The closed source RPI firmware modifies several entries in the
+device-tree that are important for proper boot. The complete list
+of these entries can be found at:
+
+https://forums.raspberrypi.com/viewtopic.php?p=1974233&hilit=firmware+device+tree+dma+ranges#p1974233
+
+The function ft_board_setup() was modified so that if u-boot has
+loaded a fdt that is different than the rpi fw fdt, then
+it will copy all those important nodes and properties
+created and modified by the rpi fw to the u-boot loaded fdt
+blob.
+
+Signed-off-by: Leonardo Amorim <lam@vizrt.com>
+---
+ board/raspberrypi/rpi/rpi.c | 273 ++++++++++++++++++++++++++++++++++++
+ 1 file changed, 273 insertions(+)
+
+diff --git a/board/raspberrypi/rpi/rpi.c b/board/raspberrypi/rpi/rpi.c
+index 17b8108cc8..f4e4a40c50 100644
+--- a/board/raspberrypi/rpi/rpi.c
++++ b/board/raspberrypi/rpi/rpi.c
+@@ -25,9 +25,12 @@
+ #endif
+ #include <watchdog.h>
+ #include <dm/pinctrl.h>
++#include <stdbool.h>
+ 
+ DECLARE_GLOBAL_DATA_PTR;
+ 
++#define MAX_LEVEL	32		/* how deeply nested we will go */
++
+ /* Assigned in lowlevel_init.S
+  * Push the variable into the .data section so that it
+  * does not get cleared later.
+@@ -262,6 +265,23 @@ static const struct rpi_model rpi_models_old_scheme[] = {
+ 	},
+ };
+ 
++/*
++ * Here's a list of node manipulations that the Rpi firmware does to proper boot
++ * the Linux Kernel.
++ * If not using the option `fw_dtb` in Kernel bootargs, the dtb passed to the
```

Comment:
fw_dtb only exists in the context of the br2rauc u-boot script so should probably not be mentioned here.  Instead, add a more generic comment about the Linux kernel needing the device-tree modifications made by the closed-source RPi firmware.

Reply:
Removed comment mentioning the `fw_dtb`. Added a more generic comment about the u-boot loaded device-tree being missing some important entries to boot the Linux Kernel when it's not using the Rpi Firmware loaded device-tree.